File: autodroid-app/app/src/main/java/com/autodroid/proxy/auth/viewmodel/SharedViewModelFactory.java
// SharedViewModelFactory.java
package com.autodroid.proxy.auth.viewmodel;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

public class SharedViewModelFactory implements ViewModelProvider.Factory {
    private static AuthViewModel sharedViewModel;

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(AuthViewModel.class)) {
            if (sharedViewModel == null) {
                sharedViewModel = new AuthViewModel();
            }
            return (T) sharedViewModel;
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: autodroid-trader-app/app/src/main/java/com/autodroid/trader/auth/viewmodel/SharedViewModelFactory.kt
// SharedViewModelFactory.kt
package com.autodroid.trader.auth.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class SharedViewModelFactory : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(AuthViewModel::class.java)) {
            if (sharedViewModel == null) {
                sharedViewModel = AuthViewModel()
            }
            return sharedViewModel as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

    companion object {
        private var sharedViewModel: AuthViewModel? = null
    }
}